datm/cime_config/namelist_definition_datm.xml

```diff
@@ -224,7 +224,7 @@
     <type>char(10)</type>
```

Comment:
Near the top of the file streamslist is being set with a setting for GSWP3 like this:

``` xml
      <value datm_mode="CLMGSWP3v1">
        CLMGSWP3v1.Solar,CLMGSWP3v1.Precip,CLMGSWP3v1.TPQW
      </value>
```

This should be expanded to have an option for GSWP3 forcing and SSP compsets. So something like:

``` xml
      <value datm_mode="CLMGSWP3v1" compset_period="SSP585">
        CLMGSWP3v1.Solar,CLMGSWP3v1.Precip,CLMGSWP3v1.TPQW,Anomaly.Forcing.cmip6.ssp585
      </value>
```

There might be a way to append just the AF file to the end, which would be preferred. But, we'd need to figure that out. The matching needs to be figured out to make sure it works right. And of course all the SSP options need to be added.

Note that compset_period needs to be added to the buildnml file which I'll sketch out below.

Reply:
Changes to buildnml:

buildnml needs to:
- Get the compset from the case
- Extract out the period part of the compset
- Send compset_period to the config object

So something like...

``` xml
    caseroot = case.get_value("CASEROOT")
    datm_mode = case.get_value("DATM_MODE")
    datm_topo = case.get_value("DATM_TOPO")
    compset = case.get_value("COMPSET")
.
.
.
    # Do some reg-ex or other magic to get the compset_period from it
    compset_period = something(compset)

    config['datm_mode'] = datm_mode
    config['datm_co2_tseries'] = datm_co2_tseries
    config['compset_period'] compset_period
.
.
.
```